### DISCUSSION TITLE
Error While Running Stokes Flow Algorithm

### DISCUSSION THREAD

**Discussion by DMaxJ** [PMEAL/OpenPNM#2389] [Q&A]
Hi, 

I am here again with another problem. Few days ago I was able to work with the installation of "dev" version of both OpenPNM and porespy. I was working with the older versions of both until 2020. Right now, I am getting some crazy behaviour when I run stokes flow algorithm. Spyder crashes, Before I can read the output on the screen, I get the following message
`The system cannot find the path specified.
OMP: Error #15: Initializing libiomp5md.dll, but found libiomp5md.dll already initialized. OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/.`

Any suggestions from your end? I am running on Windows10, Anaconda and porespy and openpnm installed in a virtual environment. Everything works fine until I apply boundary conditions. When 
`perm.run()`
is keyed in it quickly shows some output, but the error message comes before I can read it.

**Comment by ma-sadeghi** [comment]
Hi @DMaxJ Could you please run your script from terminal (or Anaconda Command Prompt in Windows)? Just run the following command in your "Anaconda Command Prompt":

```
python my_script.py
```

This way, at least you should be able to have the entire output.

**Reply by ma-sadeghi** [reply]
Also, my goto first step in these scenarios is to reproduce the error in a fresh virtualenv. Could you please do that as well?

**Reply by DMaxJ** [reply]
Thank you. I tried it from the terminal and it worked without trouble. However, with Spyder the error continues

![error](https://user-images.githubusercontent.com/24526540/165260302-e240eede-cdf4-4433-8762-79fbb9c077a5.png)

**Reply by ma-sadeghi** [reply]
Are you switching to the correct virtualenv in Spyder?

**Reply by DMaxJ** [reply]
Yes

**Comment by DMaxJ** [comment]
1. I created a fresh virtual environment and installed both openpnm and porespy. When I run the same code on Spyder, I get the same error.

2. Then I tried something else
`import os
os.environ['KMP_DUPLICATE_LIB_OK']='True'`

It seems to work. I just used it after some random internet search, and I do not think it is the right way to solve the problem
3. When I remove those two lines, I get the same error back.
4. No I have a new problem. The simulation fails when I try from the command prompt. 
![error2](https://user-images.githubusercontent.com/24526540/165275029-601d1855-4a13-43db-bf41-ba3c4f023662.PNG)

**Comment by DMaxJ** [comment]
So far the work around is 
`s.settings.default_solver = "ScipySpsolve"`
like mentioned in https://github.com/PMEAL/OpenPNM/discussions/2307

The problem seems to be with pardiso solver.

**Reply by jgostick** [reply]
Sorry for making you do so much troubleshooting for something that is a known problem:  https://github.com/PMEAL/OpenPNM/discussions/2307

**Reply by DMaxJ** [reply]
It is okay. Some times things just go crazy. Thanks.
